persist/fs/write.go

```diff
@@ -117,33 +129,41 @@ func NewWriter(opts Options) (FileSetWriter, error) {
 // Open initializes the internal state for writing to the given shard,
 // specifically creating the shard directory if it doesn't exist, and
 // opening / truncating files associated with that shard for writing.
-func (w *writer) Open(
-	namespace ident.ID,
-	blockSize time.Duration,
-	shard uint32,
-	blockStart time.Time,
-) error {
-	shardDir := ShardDirPath(w.filePathPrefix, namespace, shard)
-	if err := os.MkdirAll(shardDir, w.newDirectoryMode); err != nil {
-		return err
+func (w *writer) Open(opts WriterOpenOptions) error {
+	var shardDir string
+	var fileTimestampUnixNano time.Time
+	if opts.IsSnapshot {
+		shardDir = ShardSnapshotsDirPath(w.filePathPrefix, opts.Namespace, opts.Shard)
+		if err := os.MkdirAll(shardDir, w.newDirectoryMode); err != nil {
+			return err
+		}
+		fileTimestampUnixNano = opts.WrittenAt
```

Comment:
Couple of issues with this:
- How do you distinguish a snapshot file for two separate block starts with the same WrittenAt time? 
- This will cause filenames requiring us to read the contents of the file from disk to know what data it contains. Can we avoid that?

Reply:
discussed in person with @prateek and @r settled on using the blockstart as the name like the regular flush files, but appending a counter at the end (similar to the commit log files) to distinguish between multiple snapshots files for the same block start

Reply:
done (ended up being a big refactor to implement this properly)